osf_tests/test_node.py

```diff
@@ -3542,11 +3542,11 @@ def test_complex_template(self, auth, project, pointee, component, subproject):
         new = project.use_as_template(auth=auth)
 
         assert new.title == self._default_title(project)
-        assert len(list(new.nodes)) == len(list(project.nodes))
+        assert len(list(new.nodes)) == len(list(project.nodes)) - 1
```

Comment:
The changes you are making to this test implicitly exclude linked nodes in a way that is not obvious. 
I'd prefer if this was broken into two tests:
1. remove the pointee from `test_complex_template` and remove the special casing.
2. Add a new test with just a pointee and show that it does not get templated. 

I think that will be much simpler than arbitrarily `-1` and removing linked nodes from the data checks
If you prefer it as it is, adding notes for the special casing would work as well.